addons/mail/models/discuss/discuss_sync_mixin.py

```diff
@@ -0,0 +1,72 @@
+from collections import defaultdict
+from odoo import models, api
+from odoo.addons.web.models.models import lazymapping
+from odoo.addons.mail.tools.discuss import Store
+
+
+class DiscussSyncMixin(models.AbstractModel):
+    _name = "discuss.sync.mixin"
+    _description = "Mixin for Discuss Sync"
+
+    @api.model
+    def _sync_field_names(self):
+        """Return the field names to sync. Override in specific models if needed."""
+        return defaultdict(list)
+
+    @api.model
+    def _sync_field_handler(self):
+        return {}
+
+    def _get_field_name(self, field_description):
+        if isinstance(field_description, Store.Attr):
+            return field_description.field_name
+        return field_description
+
+    def _get_field_value(self, record, field_description):
+        """Get the value of a field based on its description."""
+        if isinstance(field_description, Store.Attr):
+            if field_description.predicate and not field_description.predicate(record):
+                return None
+            return field_description._get_value(record)
+        if isinstance(field_description, Store.Relation):
+            return field_description._get_value(record).records
+        return record[field_description]
+
+    def _get_vals(self, record):
```

Comment:
all the methods taking `record` as param should be called directly on record instead so that record = self